File: Lek.java
import java.util.Scanner;
import java.io.PrintStream;

public class Lek {
    String name;
    String mass;
    double price;
    int amount;
    void read() {
        System.getProperty("console.encoding", "utf-8");
        Scanner in = new Scanner(System.in, "utf-8");
        System.out.println("Введите название лекарства");
        name = in.nextLine();
        System.out.println("Введите код лекарства");
        mass = in.nextLine();
        System.out.println("Введите цену лекарства");
        price = in.nextDouble();
        System.out.println("Введите колличество лекарства");
        amount = in.nextInt();
    }
    void init(String mass1, String name1, double price1, int amount1){
        mass = mass1;
        name = name1;
        price = price1;
        amount = amount1;
    }
    void display(){
        System.out.println("Название лекарства:" + name);
        System.out.println("Код лекарства:" + mass);
        System.out.println("Цена лекарства:" + price);
        System.out.println("Колличество:" + amount);
    }
    void setPrice(double price1){
        price = price1;
    }
    void setAmount(int amount1){
        amount = amount + amount1;
    }
    String getMass(){
        return mass;
    }
    int getAmount(){
        return amount;
    }
}

File: Lek.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace lab8
{
    class Lek
    {
        private string mass;
        private string name;
        private double price;
        private int amount;
        public void read()
        {
            Console.WriteLine("Введите название лекарства");
            name = Console.ReadLine();
            Console.WriteLine("Введите код лекарства");
            mass = Console.ReadLine();
            do
            {
                Console.WriteLine("Введите цену лекарства");
                price = Convert.ToDouble(Console.ReadLine());
            } while (price < 0);
            do
            {
                Console.WriteLine("Введите колличество лекарства");
                amount = Convert.ToInt32(Console.ReadLine());
            } while (amount < 0);
        }
        public void init(string mass, string name, double price, int amount)
        {
            this.name = name;
            this.mass = mass;
            this.price = price;
            this.amount = amount;
        }
        public void display()
        {
            Console.WriteLine("Название товара:" + name);
            Console.WriteLine("Код лекарства:" + mass);
            Console.WriteLine("Цена лекарства:" + price);
            Console.WriteLine("Колличество:" + amount);
        }
        public int Amount
        {
            set
            {
                if (value > 0)
                {
                    amount = value;
                }
            }
            get
            {
                return amount;
            }
        }
        public double Price
        {
            set
            {
                if (value > 0)
                {
                    price = value;
                }
            }
        }
        public string Mass
        {
            get
            {
                return mass;
            }
        }
    }
}